dex/src/state.rs

```diff
@@ -871,6 +871,10 @@ impl RequestQueue<'_> {
         self.header.next_seq_num += 1;
         seq_num
     }
+
+    pub fn get_next_seq_num(& self) -> u64 {
```

Comment:
&self

Also why not just also add `generate_order_id`?

Reply:
Thoughts on this change? I just made `gen_order_id` public and added a `increment_seq: bool`. It's only used once aside from the code I'm adding to DEX.

Reply:
Thoughts on this change? I just made `gen_order_id` public and added a `increment_seq: bool`. It's only used once aside from the code I'm adding to DEX.